mutex not declared

Following a sudo apt update, sudo apt upgrade and reboot of my Pi3 I did a clean install (not sudo) and all went well until this:

2021-10-16 15:49:32 Building keogram program...
keogram.cpp:51:6: error: 'mutex' in namespace 'std' does not name a type
 std::mutex stdio_mutex;
      ^~~~~
keogram.cpp:51:1: note: 'std::mutex' is defined in header '<mutex>'; did you forget to '#include <mutex>'?
keogram.cpp:24:1:
+#include <mutex>
 
keogram.cpp:51:1:
 std::mutex stdio_mutex;
 ^~~
keogram.cpp:59:26: error: 'std::mutex' has not been declared
                     std::mutex*,       // mutex

Similar error reports followed ending with this

make[1]: *** [Makefile:128: keogram] Error 1
make[1]: Leaving directory '/home/pi/allsky/src'
make: *** [Makefile:24: all] Error 2
Compile failed!

--- a/src/keogram.cpp
+++ b/src/keogram.cpp
@@ -14,6 +14,7 @@ using namespace std;
 #include <sys/time.h>
 #include <cstdlib>
 #include <iostream>
+#include <mutex>
 #include <string>
 #include <thread>
 #include <vector>

